Realm/RLMUser.mm

```diff
@@ -323,3 +330,78 @@ - (instancetype)initUserIdentityWithProviderType:(NSString *)providerType
 }
 
 @end
+
+#pragma mark - RLMUserProfile
+
+@interface RLMUserProfile () {
+    SyncUserProfile _userProfile;
+}
+@end
+
+@implementation RLMUserProfile
+
+- (instancetype)initWithUserProfile:(SyncUserProfile)userProfile {
+    if (self = [super init]) {
+        _userProfile = userProfile;
+    }
+    return self;
+}
+
+- (NSString *)name {
+    _userProfile.data();
+    if (_userProfile.name() == util::none) {
+        return nil;
+    }
+    return @(_userProfile.name()->c_str());
```

Comment:
This is a lot of duplicated logic which could easily be extracted to a function.